src/TagsInput/TagsInput.js

```diff
@@ -11,156 +16,163 @@ import InputField from './components/InputField';
 import { StyleSheet } from '../PlatformStyleSheet';
 import { Text } from '../Text';
 
-import { INPUT_MIN_WIDTH, TAGS_MIN_WIDTH } from './constants';
-
 type Props = {|
-  +label: string,
+  +fontSize: number,
   +onChangeText: (value: string) => void,
-  +placeholder: string,
-  +selected?: string[],
+  +tags: string[],
   +disabled?: boolean,
-  +fontSize: number,
+  +label?: string,
   +onClearPress?: () => void,
+  +onKeyPress?: (e: Event) => void,
+  +placeholder?: string,
   +value?: string,
-  +autoFocus: boolean,
+  +autofocus?: boolean,
 |};
 
 type State = {
   value: string,
   isFocus: boolean,
-  containerWidth: ?number,
-  inputWidth: ?number,
 };
 
-export default class TagInput extends React.Component<Props, State> {
-  inputRef: ?{ current: null | React.Element<typeof TextInput> };
+type Event = { nativeEvent: { key: string } };
+
+const boxShadow = isFocus => {
+  if (Platform.OS === 'web') {
+    return {
+      boxShadow: isFocus
+        ? `${defaultTokens.borderColorInputFocus} 0 0 0 2px inset`
+        : `${defaultTokens.borderColorInput} 0 0 0 1px inset`,
+    };
+  }
+  return null;
+};
+
+export default class TagsInput extends React.Component<Props, State> {
+  inputRef: any;
+  scrollRef: any;
+
   static defaultProps = {
-    fontSize: 16,
-    autoFocus: true,
+    fontSize: parseFloat(defaultTokens.fontSizeButtonLarge),
+    tags: [],
   };
 
   constructor(props: Props) {
     super(props);
     this.inputRef = React.createRef();
+    this.scrollRef = React.createRef();
 
     this.state = {
-      value: props.value ?? '',
-      containerWidth: null,
-      inputWidth: null,
-      isFocus: props.autoFocus,
```

Comment:
Likewise here.

Reply:
`isFocus: props.autofocus || false,` I will change it like this so we don't need a set default value. Or you still prefer to have it?

Reply:
Well, we have `defaultProps` anyway, so I would put it there. Otherwise, we have default values spread in different places.

Reply:
OK

Reply:
make sense